extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/DevConsoleProcessor.java

```diff
@@ -155,7 +157,17 @@ public void handle(HttpServerRequest event) {
 
     }
 
-    protected static void newRouter(Engine engine) {
+    protected static void newRouter(Engine engine,
+            HttpRootPathBuildItem httpRootPathBuildItem,
+            NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem) {
+
+        // "" or "/myroot" 
+        String httpRootPath = httpRootPathBuildItem.adjustPath("/");
+        httpRootPath = httpRootPath.substring(0, httpRootPath.lastIndexOf("/"));
+        // "" or "/myroot" or "/q" or "/myroot/q"
+        String frameworkRootPath = httpRootPathBuildItem.adjustPath(nonApplicationRootPathBuildItem.adjustPath("/"));
```

Comment:
Are you absolutely sure that those will have a trailing `/`? There's no way for the user to set them to `/foo`?

Reply:
Yes, adjustPath add that

Reply:
(I tested with /foo)

Reply:
OK great.